feat: Add PS256, PS384 and PS512 support

Andrew-Lees11 opened the pull request:
This pull request adds two now signers and verifiers for the PS256 and PS384 JWT algorithms. 
These use the new RSA-PSS support introduced in BlueRSA 1.0.31.

Tests have been added for sign -> verify cycle and against the JWT.io implementation by using a JWT they generated and generating a JWT that is then verfied on JWT.io.

This adds the functionality requested by issue #55.

djones6 commented on Tests/SwiftJWTTests/TestJWT.swift:
```diff
@@ -241,6 +249,59 @@ class TestJWT: XCTestCase {
+            0zrEMBVFpW7RroTx5qdZAQVNB7b5S/iucBDy5pvMAre3tJ30mhpt0tZtLOL7tpphh
+            jf4FRnHS/VHb8wN6mrUu4r+7v2ZpfzJwQbkDBjxzyCl0iDGQoQo0dMUIuE0yui9ur
+            9+9L5gEREuDPMRXNPIwh/dgMEWo7QIDAQAB
+            -----END RSA PUBLIC KEY-----
```
Could we update the RSA key that the tests use to be 2048-bit instead?

Andrew-Lees11 replied:
Done

Andrew-Lees11 commented:
I have updated our RSA key to be 2048 bit so that it can be used for ps512. I have also added if available calls to our tests since RSA-PSS requires MacOS 10.13.
